packages/debug/src/node/vscode/vscode-debug-adapter-contribution.ts

```diff
@@ -229,12 +229,22 @@ export abstract class AbstractVSCodeDebugAdapterContribution implements DebugAda
         if (runtime && runtime.indexOf('./') === 0) {
             runtime = path.join(this.extensionPath, runtime);
         }
+
         const runtimeArgs = info && info.runtimeArgs || contribution.runtimeArgs || [];
-        const command = runtime ? runtime : program;
-        const args = runtime ? [...runtimeArgs, program, ...programArgs] : programArgs;
-        return {
-            command,
-            args
-        };
+        if (runtime === 'node') {
+            const modulePath = program;
+            const args = [...runtimeArgs, ...programArgs];
```

Comment:
You will need a bit more changes.

`args` will be passed to a command like:

```sh
node modulePath ...runtimeArgs ...programArgs
```

When we want:

```sh
node ...runtimeArgs modulePath ...programArgs
```

I suggest modifying:

```ts
export interface DebugAdapterForkExecutable {
    modulePath: string;
    execArgs?: string[];
    args?: string[];
}
```

And then make the necessary modifications in the spawning pipeline to correctly pass arguments, as it doesn't seem to be supported currently.